lib/beaker/testmode_switcher/beaker_runners.rb

```diff
@@ -124,7 +129,12 @@ def execute_manifest_on(hosts, manifest, opts = {})
           expect_failures: true,
           acceptable_exit_codes: (0...256))
 
-        handle_puppet_run_returned_exit_code(get_acceptable_puppet_run_exit_codes(opts), res.exit_code)
+        if res.is_a? Array
+          res.each_with_index {|val, index|
+            handle_puppet_run_returned_exit_code(get_acceptable_puppet_run_exit_codes(opts), res[index].exit_code)}
```

Comment:
As above, Shouldn't it be `val.exit_code`

Also means you don't need the index. so `res.each` should be fine